src/appengine/handlers/cron/triage.py

```diff
@@ -250,11 +254,117 @@ def _check_and_update_similar_bug(testcase, issue_tracker):
   return False
 
 
-def _file_issue(testcase, issue_tracker):
+def _query_job_bugs_filed_count(job_type, valid_timestamp):
+  """Gets the number of bugs that have been filed for a given job
+  within a given time period."""
+  return data_types.FiledBug.query(
+      data_types.FiledBug.job_type == job_type and
+      data_types.FiledBug.timestamp >= valid_timestamp).count()
+
+
+def _query_project_bugs_filed_count(project_name, valid_timestamp):
+  """Gets the number of bugs that have been filed for a given project
+  within a given time period."""
+  return data_types.FiledBug.query(
+      data_types.FiledBug.project_name == project_name and
+      data_types.FiledBug.timestamp >= valid_timestamp).count()
+
+
+def _get_job_bugs_filing_max(job_type, bug_filing_max_24_hours_per_job):
+  """Gets the maximum number of bugs that can be filed for a given job."""
+  if job_type in bug_filing_max_24_hours_per_job:
+    return bug_filing_max_24_hours_per_job[job_type]
+
+  max_bugs = None
+  job = data_types.Job.query(data_types.Job.name == job_type).get()
+  if job and 'BUG_FILING_MAX_24_HOURS_PER_JOB' in job.get_environment():
+    try:
+      max_bugs = int(job.get_environment()['BUG_FILING_MAX_24_HOURS_PER_JOB'])
+    except Exception:
+      logs.log_error(
+          'Invalid environment value of \'BUG_FILING_MAX_24_HOURS_PER_JOB\' '
+          f'for job type {job_type}.')
+
+  bug_filing_max_24_hours_per_job[job_type] = max_bugs
+  return max_bugs
+
+
+def _get_project_bugs_filing_max(bug_filing_max_24_hours_per_project):
+  """Gets the maximum number of bugs that can be filed per project."""
+  if len(bug_filing_max_24_hours_per_project) != 0:
+    return bug_filing_max_24_hours_per_project[0]
+  issue_tracker_config = local_config.IssueTrackerConfig()
+  config = issue_tracker_config.get(data_handler.get_issue_tracker_name())
+  try:
+    bug_filing_max_24_hours_per_project.append(
+        int(
+            config.get('bug_filing_max_24_hours_per_project',
```

Comment:
nit: soomething  like "max_bugs_per_project_per_24hrs" is slightly shorter and better.